src/services/studio/studio.service.ts

```diff
@@ -181,24 +182,72 @@ class StudioService {
+
+                    if (studioManifest) {
+                        const spaceId = await this.findDesiredSpaceIdForPackage(studioManifest);
+                        studioManifest.space.id = spaceId;
+
+                        const packageZip = new AdmZip(file.getData());
+                        packageZip.getEntries().forEach(nodeFile => {
+                            if (nodeFile.entryName.endsWith(BatchExportImportConstants.YAML_EXTENSION)) {
+                                const updatedNodeFile = this.updateSpaceIdForNode(nodeFile.getData().toString(), spaceId);
+                                packageZip.updateFile(nodeFile, Buffer.from(updatedNodeFile));
+                            }
+                        });
+                        exportedFiles.updateFile(file, packageZip.toBuffer());
+                    }
+                }
+            }
+        }
+        return exportedFiles;
+    }
+
+    private isStudioPackage(studioManifests: StudioPackageManifest[], packageKey: string): boolean {
+        return studioManifests.some(manifest => manifest.packageKey === packageKey);
+    }
 
+    private async findDesiredSpaceIdForPackage(studioPackageManifest: StudioPackageManifest): Promise<string> {
         const allSpaces = await spaceService.refreshAndGetAllSpaces();
-        let targetSpace = allSpaces.find(space => space.name === manifest.space.name);
 
-        if (!targetSpace) {
-            targetSpace = await spaceService.createSpace(manifest.space.name, manifest.space.iconReference);
+        if (studioPackageManifest.space.id) {
+            const targetSpace = allSpaces.find(space => space.id === studioPackageManifest.space.id);
+           if (!targetSpace) {
+                throw Error("Provided space ID does not exist.");
+            }
+            return targetSpace.id;
+        }
+
+        const targetSpaceByName = allSpaces.find(space => space.name === studioPackageManifest.space.name);
+        if (targetSpaceByName) {
+            return targetSpaceByName.id;
         }
 
-        await packageApi.movePackageToSpace(nodeInTargetTeam.id, targetSpace.id);
+        const spaceTransport = await spaceService.createSpace(studioPackageManifest.space.name, studioPackageManifest.space.iconReference);
+        return spaceTransport.id;
     }
 
     private async assignRuntimeVariables(manifest: StudioPackageManifest): Promise<void> {
         if (manifest.runtimeVariableAssignments.length) {
             await variableService.assignVariableValues(manifest.packageKey, manifest.runtimeVariableAssignments);
         }
     }
+    private updateSpaceIdForNode(nodeContent: string, spaceId: string): string {
+        const exportedNode: NodeExportTransport = parse(nodeContent);
+        const oldSpaceId = exportedNode.unversionedMetadata.spaceId;
+
+        nodeContent = nodeContent.replace(new RegExp(oldSpaceId, "g"), spaceId);
```

Comment:
q: Is there a reason to use a regex instead of `replaceAll` here?

Reply:
replaceAll is available from version 15 and the min required version to use Content CLI is >10, if we were to update this to 15 it requires announcing it as a change to all the user (studio personas).